symengine/cwrapper.cpp

```diff
@@ -330,29 +331,29 @@ int real_mpfr_is_zero(const basic s)
 }
 
 #endif // HAVE_SYMENGINE_MPFR
-#ifdef HAVE_SYMENGINE_MPC
-int complex_mpc_is_zero(const basic s)
-{
-    SYMENGINE_ASSERT(is_a<ComplexMPC>(*(s->m)));
-    return (int)((down_cast<const ComplexMPC &>(*(s->m))).is_zero());
-}
-
-CWRAPPER_OUTPUT_TYPE complex_mpc_real_part(basic s, const basic com)
+        
+CWRAPPER_OUTPUT_TYPE complex_base_real_part(basic s, const basic com)
 {
     CWRAPPER_BEGIN
-    SYMENGINE_ASSERT(is_a<ComplexMPC>(*(com->m)));
-    s->m = (down_cast<const ComplexMPC &>(*(com->m))).real_part();
+    SYMENGINE_ASSERT(is_a<ComplexBase>(*(com->m)));
```

Comment:
`is_a<ComplexBase>` checks exact type. Use `is_a_Complex` for checking that it is complex.